src/Support/ResolveInfoFieldsAndArguments.php

```diff
@@ -132,15 +143,67 @@ private function foldSelectionSet(SelectionSetNode $selectionSet, int $descend):
         return $fields;
     }
 
-    private function getValue(ArgumentNode $argumentNode)
+    /**
+     * @param ValueNode $value
+     * @return mixed
+     * @throws \Exception
+     */
+    private function getValue(ValueNode $value)
     {
-        $value = $argumentNode->value;
         if ($value instanceof VariableNode) {
             $variableName = $value->name->value;
 
             return $this->info->variableValues[$variableName] ?? null;
         }
+        // Scalar Types
+        if ($value instanceof IntValueNode || $value instanceof FloatValueNode || $value instanceof StringValueNode || $value instanceof BooleanValueNode || $value instanceof EnumValueNode) {
+            return $value->value;
+        }
+        // null Type
+        if ($value instanceof NullValueNode) {
+            return;
+        }
+        // object Type
+        if ($value instanceof ObjectValueNode) {
+            return $this->getInputObjectValue($value);
+        }
+        // list object Type
+        if ($value instanceof ListValueNode) {
+            return $this->getInputListObjectValue($value);
+        }
+
+        throw new \Exception('Failed to resolve unknown ValueNode type');
+    }
+
+    /**
+     * @param ObjectValueNode $objectValueNode
+     * @return array
+     * @throws \Exception
+     */
+    private function getInputObjectValue(ObjectValueNode $objectValueNode)
+    {
+        $value = [];
+        foreach ($objectValueNode->fields->getIterator() as $item) {
+            if ($item instanceof ObjectFieldNode) {
```

Comment:
Not sure about this.

The code makes it look like `$item` could be something else. But reading `\GraphQL\Language\AST\ObjectValueNode::$fields` it's clear this is not intended.

How can we improve this? Should we remove the check? Or add an `} else {` and throw an exception?

Because as it is right now, if that case ever would happen, we silently would carry on.

Reply:
1. Currently, item only handles objects of type ObjectFieldNode
2. If the current item object contains other scalar types or listNode and so on, the resolution continues recursively